qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1753,6 +1756,9 @@ def _on_url_changed(self, url: QUrl) -> None:
         Since update_for_url() is idempotent, it doesn't matter much if we end
         up doing it twice.
         """
+        original = urlutils.extract_error_url(url)
+        if original:
+            url = original[0]
```

Comment:
Why only for QtWebEngine? This should probably be done in `BrowserTab._on_url_changed` instead - especially because settings set for the original URL probably should not be active for the error page (?).

Reply:
Good point. Edited.

On Webkit, JavaScript still doesn't work on the error page if the real page has JavaScript disabled. (although I think it has been like that before this pull request)